### ISSUE TITLE
Support `multiline` appearance

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#6337]
Enketo supports a `multiline` appearance which shows a multiline textbox. It also supports the `rows` attribute the same way as Collect does to specify a minimum number of lines that a textbox should be shown as having.

Collect should support `multiline`. This should behave like the ["text areas"](https://m3.material.io/components/text-fields/guidelines#caf14aa9-26ce-48cc-9076-3a383a9ad679) Material text input and have a height of 4 lines (unless the question has the `rows` parameter in which case that number should be used).

Related: https://github.com/getodk/collect/issues/2215, https://forum.getodk.org/t/odk-collect-text-entry-options-limited-by-single-line-design/37942/5

**Comment by lognaturel** [comment]
Carriage returns can cause issues with CSV exports from Central: https://forum.getodk.org/t/csv-data-export-with-carriage-returns-is-badly-formatted/47980

We've started to discuss the possibility of introducing the `multiline` appearance and eventually not allowing carriage returns in text fields without that appearance.

**Comment by lognaturel** [comment]
I think this is also our path to addressing https://github.com/getodk/collect/issues/6665

**Comment by seadowg** [comment]
> Collect should support multiline. We need to decide how many rows it should display by default and whether it should expand horizontally or scroll.

@alyblenkin @lognaturel what are we thinking on this? Obviously we'll get feedback on it, but I feel like we can make a gut call now and tweak later.

**Comment by lognaturel** [comment]
Sorry I meant expand vertically, not horizontally!

My gut is 5 rows and expand vertically because I think expanding is more Android-typical. @alyblenkin?

**Comment by alyblenkin** [comment]
I'm very glad we aren't expanding horizontally!

And yes, Material suggests that [multi-line text fields grow](https://m3.material.io/components/text-fields/guidelines#caf14aa9-26ce-48cc-9076-3a383a9ad679) to accommodate multiple lines of text but it's still a fixed height. 

"Text areas are taller than text fields and wrap overflow text onto a new line. They are a fixed height and scroll vertically when the cursor reaches the bottom of the field. The large initial size indicates that longer responses are possible and encouraged. These should be used instead of multi-line fields on the web. Ensure the height of a text area fits within mobile screen sizes."

**Comment by lognaturel** [comment]
Oh! I find "These should be used instead of multi-line fields on the web" a bit confusing. Sounds like even in an Android app we should pick a number of rows for a fixed height and scroll vertically?

**Comment by alyblenkin** [comment]
Yes, I think so. We should play around with this but I'm assuming fixing the height to 3-4 rows and then the user can scroll vertically within the text area should work pretty well.

**Comment by seadowg** [comment]
> Yes, I think so. We should play around with this but I'm assuming fixing the height to 3-4 rows and then the user can scroll vertically within the text area should work pretty well.

Let's try 4 lines for the moment and we can see how that goes.
